### PR TITLE
[UPDATES] New stubs for Automatic Updates service and WUSA.exe.

### PR DESCRIPTION
**Pull request by Getequ**
## Purpose

Presence of this modules required for installation by some NT6+ applications.

## Proposed changes

This stubs do nothing, but help to continue installation process

### CODE REVIEW

File: base/applications/CMakeLists.txt [hunk 1]
```diff
@@ -42,3 +42,4 @@ add_subdirectory(winhlp32)
 add_subdirectory(winver)
 add_subdirectory(wordpad)
 add_subdirectory(write)
+add_subdirectory(wusa)
```
Comment (HBelusca) [inline base/applications/CMakeLists.txt]: Please add an extra ending newline.

File: base/services/CMakeLists.txt [hunk 2]
```diff
@@ -15,3 +15,4 @@ add_subdirectory(umpnpmgr)
 add_subdirectory(wkssvc)
 add_subdirectory(wlansvc)
 add_subdirectory(wmisvc)
+add_subdirectory(wuauserv)
```
Comment (HBelusca) [inline base/services/CMakeLists.txt]: Please add an extra ending newline.

File: base/services/wuauserv/wuauserv.c [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+*
+*  This program is distributed in the hope that it will be useful,
+*  but WITHOUT ANY WARRANTY; without even the implied warranty of
+*  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+*  GNU General Public License for more details.
+*
+*  You should have received a copy of the GNU General Public License along
+*  with this program; if not, write to the Free Software Foundation, Inc.,
+*  51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+*/
```
Comment (HBelusca) [inline base/services/wuauserv/wuauserv.c]: That stuff should not be needed now.

File: base/services/wuauserv/wuauserv.c [hunk 4]
```diff
@@ -0,0 +1,160 @@
...
+/*
+* COPYRIGHT:        See COPYING in the top level directory
+* PROJECT:          ReactOS Services
+* FILE:             base/services/wuauserv/wuauserv.c
+* PURPOSE:          Windows Update service stub.
+*                   Stub is required for some application at the installation phase.
+* PROGRAMMER:       Denis Malikov
+*/
```
Comment (HBelusca) [inline base/services/wuauserv/wuauserv.c]: Please use the new code header format (see https://www.reactos.org/wiki/Coding_Style ), respecting the whitespace indentation.
Note the presence of the license line, which basically replaces the block of text from above.

File: base/services/wuauserv/wuauserv.c [hunk 5]
```diff
@@ -0,0 +1,160 @@
...
+        DPRINT1("RegisterServiceCtrlHandlerExW() failed! (Error %lu)\n", GetLastError());
+        return;
+    }
+
+    UpdateServiceStatus(SERVICE_RUNNING);
+
+    do
+    {
+        Sleep(1);
+    } while (1);
```
Comment (HBelusca) [inline base/services/wuauserv/wuauserv.c]: Please use WaitForSingleObject instead and/or spawn a separate thread that just waits for the SCM to send a stop message.

File: base/applications/wusa/resource.rc [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ */
```
Comment (HBelusca) [inline base/applications/wusa/resource.rc]: Use again the new code headers instead.

File: base/services/wuauserv/wuauserv.c (hunk 3 above)
Comment (ThFabba) [reply base/services/wuauserv/wuauserv.c]: There's nothing wrong with a standard license header if the author prefers it. Although if there's no preference, the shortened SPDX version saves some space.

File: base/applications/wusa/wusa.c [hunk 7]
```diff
@@ -0,0 +1,14 @@
+#define WIN32_NO_STATUS
```
Comment (HBelusca) [inline base/applications/wusa/wusa.c]: Please add a code header.

**Comment by ThFabba** [comment]
Do these stubs by themselves already fix something, or do you have plans to do further development on these components?
I ask because if they're going to remain stubs, we should probably hold off on adding them to the bootcd/registry to save disk space/memory/bootup time until needed.

File: base/applications/wusa/wusa.c [hunk 8]
```diff
@@ -0,0 +1,14 @@
...
+#define WIN32_LEAN_AND_MEAN
+#include <windows.h>
+#include <tchar.h>
+
+int
+WINAPI
+_tWinMain(HINSTANCE hCurInst,
```
Comment (ThFabba) [inline base/applications/wusa/wusa.c]: You can make this wWinMain for simplicity. TCHARs aren't really very useful these days.

File: base/services/wuauserv/wuauserv.c (hunk 3 above)
Comment (ThFabba) [reply base/services/wuauserv/wuauserv.c]: Oh, I didn't see this in context. Yes, of course you only need either/or, not both types of headers.

File: base/services/wuauserv/CMakeLists.txt [hunk 9]
```diff
@@ -0,0 +1,10 @@
+spec2def(wuauserv.dll wuauserv.spec ADD_IMPORTLIB)
```
Comment (ThFabba) [inline base/services/wuauserv/CMakeLists.txt]: We probably won't need an import lib for this dll.

File: boot/bootdata/hivesft.inf [hunk 10]
```diff
@@ -1653,7 +1653,7 @@ HKLM,"SOFTWARE\Microsoft\Ole","EnableRemoteConnect",0x00000000,"N"
 ; SvcHost services
 HKLM,"SOFTWARE\Microsoft\Windows NT\CurrentVersion\SvcHost",,0x00000012
 HKLM,"SOFTWARE\Microsoft\Windows NT\CurrentVersion\SvcHost","DcomLaunch",0x00010000,"PlugPlay"
-HKLM,"SOFTWARE\Microsoft\Windows NT\CurrentVersion\SvcHost","netsvcs",0x00010000,"DHCP","BITS","lanmanserver","lanmanworkstation","Schedule","Themes","winmgmt"
+HKLM,"SOFTWARE\Microsoft\Windows NT\CurrentVersion\SvcHost","netsvcs",0x00010000,"DHCP","BITS","lanmanserver","lanmanworkstation","Schedule","Themes","winmgmt","wuauserv"
```
Comment (ThFabba) [inline boot/bootdata/hivesft.inf]: This should likely remain commented out until the service is actually used by something, as per my other comment.

File: boot/bootdata/hivesys.inf [hunk 11]
```diff
@@ -211,6 +212,7 @@ HKLM,"SYSTEM\CurrentControlSet\Control\SafeBoot\Network\vga.sys","",0x00000000,"
 HKLM,"SYSTEM\CurrentControlSet\Control\SafeBoot\Network\vgasave.sys","",0x00000000,"Driver"
 HKLM,"SYSTEM\CurrentControlSet\Control\SafeBoot\Network\WinMgmt","",0x00000000,"Service"
 HKLM,"SYSTEM\CurrentControlSet\Control\SafeBoot\Network\WZCSVC","",0x00000000,"Service"
+HKLM,"SYSTEM\CurrentControlSet\Control\SafeBoot\Network\wuauserv","",0x00000000,"Service"
```
Comment (ThFabba) [inline boot/bootdata/hivesys.inf]: This service doesn't need to run in safe mode (and especially not safe mode without networking)

File: boot/bootdata/hivesft.inf (hunk 10 above)
Comment (Getequ) [reply boot/bootdata/hivesft.inf]: @ThFabba i fixed locally all comments except this. Without this string service does not start (failed with 1060). Is this block from merging?

Comment (ThFabba) [reply boot/bootdata/hivesft.inf]: Not starting the service is exactly what I was going for! :) (hence less memory use/boot time)
However if apps depend on the service running and not just the files existing, then of course you'll have to leave it in here.

Comment (Getequ) [reply boot/bootdata/hivesft.inf]: Ok, let's leave it not starting for now. From times to times i tring to install .NET 4.5.1 :)

**Comment by Getequ** [comment]
@binarymaster ok, add this email also

**Comment by Getequ** [comment]
Oops, little screwed up with my master. Recreate a PR

**Comment by binarymaster** [comment]
Why not just `git push --force-with-lease`?

**Comment by Getequ** [comment]
didn't know about this (

**Comment by HBelusca** [comment]
Then you can reopen the PR ^^

**Comment by Getequ** [comment]
i trying to execute this command after binarymaster said about that, but git says that nothing to push

**Pull request closed without merging**